src/openrct2/world/park.c

```diff
@@ -949,8 +949,13 @@ static money32 map_buy_land_rights_for_tile(sint32 x, sint32 y, sint32 setting,
 
         rct_map_element* mapElement = map_get_first_element_at(x / 32, y / 32);
         do {
-            if (map_element_get_type(mapElement) == MAP_ELEMENT_TYPE_ENTRANCE){
-                return 0;
+            if (map_element_get_type(mapElement) == MAP_ELEMENT_TYPE_ENTRANCE) {
+                uint8 _newOwnership = (flags & 0xFF00) >> 4;
+                // Allow construction rights available / for sale on park entrances on surface.
+                if ((_newOwnership != 0 && _newOwnership != 16 && _newOwnership != 64) ||
```

Comment:
16 and 64 appear to be magic numbers. I expect there are defines somewhere you can use.